### DISCUSSION TITLE
Intellisense for Svelte Projects in Zed Code Editor

### DISCUSSION THREAD

**Discussion by ChristianPerez34** [zed-industries/zed#6619] [Show and tell]
Hello Zed community,

For all users interested in working on Svelte projects using the Zed code editor, here's an important step that will enhance your workflow. To get intellisense working in your coding environment, kindly add the following settings JSON block into your Zed code editor settings file:

```JSON
{
  "lsp": {
    "typescript-language-server": {
      "initialization_options": {
        "preferences": {
          "includeInlayParameterNameHints": "all",
          "includeInlayParameterNameHintsWhenArgumentMatchesName": true,
          "includeInlayFunctionParameterTypeHints": true,
          "includeInlayVariableTypeHints": true,
          "includeInlayVariableTypeHintsWhenTypeMatchesName": false,
          "includeInlayPropertyDeclarationTypeHints": true,
          "includeInlayFunctionLikeReturnTypeHints": true,
          "includeInlayEnumMemberValueHints": true,
          "includeInlayEnumMemberDeclarationTypes": true
        }
      }
    }
  }
}
```

By incorporating these settings, I was able to successfully get the Intellisense working in the Zed code editor for my Svelte projects. I'm sharing these settings with you with the hope that it will aid your development process too. 

Happy coding!

**Comment by JosephTLyons** [comment]
Hey @ChristianPerez34, would you want to open a PR to add this to our docs, by adding a new section similar to the existing ones? If not, I can do it, I just wanted to make sure you had the opportunity.

- https://github.com/zed-industries/docs/blob/main/configuration/configuring-zed.md#rust

**Reply by ChristianPerez34** [reply]
Sure thing!

**Reply by JosephTLyons** [reply]
Appreciate it! :)

**Reply by iambudi** [reply]
Is there a way to format .svelte file?
Currently i manually use `npx prettier --write [file.svelte or .]`
Thinking if it's possible to bind it as cmd+shift+i
